File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/transport/StreamingTransportExecutor.java
package org.hyperledger.iroha.android.client.transport;

import java.util.concurrent.CompletableFuture;

/**
 * Transport executor that can surface streaming responses without buffering the full payload.
 */
public interface StreamingTransportExecutor extends TransportExecutor {

  /**
   * Executes the request and returns a streaming response once headers are available.
   */
  CompletableFuture<TransportStreamResponse> openStream(TransportRequest request);
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/transport/StreamingTransportExecutor.kt
package org.hyperledger.iroha.sdk.client.transport

import java.util.concurrent.CompletableFuture

/**
 * Transport executor that can surface streaming responses without buffering the full payload.
 */
interface StreamingTransportExecutor : TransportExecutor {

    /**
     * Executes the request and returns a streaming response once headers are available.
     */
    fun openStream(request: TransportRequest): CompletableFuture<TransportStreamResponse>
}
